osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/infra/networking/DisposableNioEventLoopGroup.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2020 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.shared.infra.networking;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.BeanNameAware;
+import org.springframework.beans.factory.DisposableBean;
+
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.util.concurrent.Future;
+
+public class DisposableNioEventLoopGroup extends NioEventLoopGroup implements BeanNameAware, DisposableBean {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DisposableNioEventLoopGroup.class);
+
+    private String name = "disposableNioEventLoopGroup";
+
+    public DisposableNioEventLoopGroup() {
+        super();
+    }
+
+    @Override
+    public void destroy() {
+        LOGGER.info("Destroying DisposableNioEventLoopGroup Bean: {}", this.name);
+        final Future<?> f = this.shutdownGracefully();
+        try {
+            f.await();
+        } catch (final InterruptedException e) {
+            throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS00112)